File: routes/projetos.py
from flask import request, jsonify


def register_projetos_routes(app, lista_de_projetos):
    @app.route('/projetos', methods=['GET'])
    def api_list_projetos():
        print("Projetos listados com sucesso")
        print(lista_de_projetos)
        return jsonify(lista_de_projetos)

    @app.route('/projetos', methods=['POST'])
    def api_create_projeto():
        data = request.get_json() or {}
        nome = data.get('nome')
        tipo = data.get('tipo')
        link = data.get('link')
        if nome and tipo and link:
            lista_de_projetos.append({'nome': nome, 'tipo': tipo, 'link': link})
            print("Projeto postado com sucesso")
            print(lista_de_projetos)
            return jsonify({'success': True}), 201
        return jsonify({'error': 'invalid data'}), 400

    @app.route('/projetos/<int:index>', methods=['PUT'])
    def api_update_projeto(index):
        if 0 <= index < len(lista_de_projetos):
            data = request.get_json() or {}
            projeto = lista_de_projetos[index]
            projeto['nome'] = data.get('nome', projeto.get('nome'))
            projeto['tipo'] = data.get('tipo', projeto.get('tipo'))
            projeto['link'] = data.get('link', projeto.get('link'))
            print("Projeto editado com sucesso")
            print(lista_de_projetos)
            return jsonify({'success': True})
        return jsonify({'error': 'not found'}), 404

    @app.route('/projetos/<int:index>', methods=['DELETE'])
    def api_delete_projeto(index):
        if 0 <= index < len(lista_de_projetos):
            lista_de_projetos.pop(index)
            print("Projeto excluído com sucesso")
            print(lista_de_projetos)
            return jsonify({'success': True})
        return jsonify({'error': 'not found'}), 404

File: routes/projetos.js
const express = require("express");
const router = express.Router();
const { PrismaClient } = require("@prisma/client");

const prisma = new PrismaClient();

// LISTAR
router.get("/", async (req, res) => {
    const projetos = await prisma.projeto.findMany();
    res.json(projetos);
});

// CRIAR
router.post("/", async (req, res) => {
    const { nome, tipo, link } = req.body;

    const projeto = await prisma.projeto.create({
        data: { nome, tipo, link }
    });

    res.status(201).json(projeto);
});

// EDITAR (USANDO ID)
router.put("/:id", async (req, res) => {
    const id = Number(req.params.id);

    try {
        const atualizado = await prisma.projeto.update({
            where: { id },
            data: {
                nome: req.body.nome,
                tipo: req.body.tipo,
                link: req.body.link
            }
        });

        res.json(atualizado);
    } catch (err) {
        res.status(404).json({ error: "Projeto não encontrado" });
    }
});

// DELETAR (USANDO ID)
router.delete("/:id", async (req, res) => {
    const id = Number(req.params.id);

    try {
        await prisma.projeto.delete({
            where: { id }
        });

        res.json({ success: true });
    } catch (err) {
        res.status(404).json({ error: "Projeto não encontrado" });
    }
});

module.exports = router;
